python/cuml/common/input_utils.py

```diff
@@ -403,6 +403,28 @@ def input_to_cupy_array(X,
     Identical to input_to_cuml_array but it returns a cupy array instead of
     CumlArray
     """
+    if isinstance(X, cudf.DataFrame):
```

Comment:
I guess the reason we want this is because things like preprocessing and similar can accept `nan`, right? But the majority of estimators (all?) cannot accept nans, so maybe it would be benefitial to introduce a new parameter `fail_on_nan` so that the caller can request to fail here for the models that cannot take `nan` fail here as opposed to later?

Reply:
Thank you for the review. I created a `fail_on_nan` parameter. It is set to True by default in `input_to_cupy_array` and to False in the preprocessing adapter. In the preprocessing adapter (`check_array` function), multiple checks are done thanks to the `force_all_finite` parameter.